client/scripts/vite-plugin-yaml.js

```diff
@@ -0,0 +1,25 @@
+const yaml = require("js-yaml")
+
+const fileRegex = /\.(yaml|yml)$/
+
+module.exports = function viteYAMLPlugin() {
+  return {
+    name: "yaml",
+
+    transform(src, id) {
+      if (fileRegex.test(id)) {
+        const obj = yaml.load(src)
+        if (!obj || typeof obj !== "object") {
+          throw new Error(
+            "Invalid YAML provided to plugin. YAML provided must resolve to an object."
+          )
+        }
+
+        return {
+          code: `export default JSON.parse(${JSON.stringify(JSON.stringify(obj))});`,
```

Comment:
This is interesting, why does it need to be stringified twice? I guess because the entire file is a JSON string that has a JSON object? Either way, I would suggest leaving a comment explaining this here.

Reply:
I actually don't know precisely why it's like this, because this technique is copied from how Vite does it internally for JSON files.